addons/stock/models/stock_move.py

```diff
@@ -188,6 +190,12 @@ def _compute_is_locked(self):
             if move.picking_id:
                 move.is_locked = move.picking_id.is_locked
 
+    @api.depends('move_line_ids')
+    def _compute_number_to_generate(self):
+        for move in self:
+            filtered_move_line = self.move_line_ids.filtered(lambda l: not l.lot_name and not l.lot_id)
+            move.number_to_generate = len(filtered_move_line)
```

Comment:
search_count?

Reply:
even a read_group outside the for and use __count

Reply:
so just make next_serial_count a classic integer field an set its value after _action_assign